src/util/options.rs

```diff
@@ -235,27 +235,36 @@ macro_rules! options {
                     _ => panic!("Invalid Options key: {}", s)
                 }
             }
-        }
-        impl Default for Options {
-            fn default() -> Self {
-                let mut options = Options {
-                    $($name: MMTKOption::new($default, $validator, $env_var,$command_line)),*
-                };
 
-                // If we have env vars that start with MMTK_ and match any option (such as MMTK_STRESS_FACTOR),
-                // we set the option to its value (if it is a valid value). Otherwise, use the default value.
+            /// Create an `Options` instance with built-in default settings.
+            pub fn new() -> Self {
```

Comment:
The public API changes include this function.
```console
Added items to the public API
=============================
+pub fn mmtk::util::options::Options::new() -> Self
+pub fn mmtk::util::options::Options::read_env_var_settings(&mut self)
+pub fn mmtk::MMTKBuilder::new_no_env_vars() -> Self
```

But this function does not need to public. The other two are fine.

Reply:
Done.  I have made `new()` private.

`Options` still implements `Default`, so VMs can still instantiate it, although `Options` should usually be created by the `MMTKBuilder`.